src/vos/vos_iterator.c

```diff
@@ -490,25 +488,21 @@ reset_anchors(vos_iter_type_t type, struct vos_iter_anchors *anchors)
 {
 	switch (type) {
 	case VOS_ITER_DKEY:
-		D_ASSERT(daos_anchor_is_eof(&anchors->ia_dkey));
```

Comment:
Why these asserts are removed?

Reply:
Normally, when we call reset_anchors(), the former akey level iteration should have reached the end. But here, we exist former akey level iteration may because we hit some 'prepared' DTX, and then because we want to detect more 'prepared' DTX to improve the efficiency when check with leader, then we try to iterator a bit ahead.